### ISSUE TITLE
schema upgrade to 2.11.0.sql  failed on line 2

### ISSUE THREAD

**Issue by tomabg** [Icinga/icingaweb2#4863]
schema Upgrade with /usr/share/doc/icingaweb2/schema/mysql-upgrades/2.11.0.sql fails on line 2

# mysql -V
mysql  Ver 15.1 Distrib 5.5.68-MariaDB, for Linux (x86_64) using readline 5.1
# mysql -u root -p icingaweb
Enter password:
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 839
Server version: 5.5.68-MariaDB MariaDB Server

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [icingaweb]> ALTER TABLE `icingaweb_group` CONVERT TO CHARACTER SET utf8mb4;
Query OK, 0 rows affected (0.05 sec)
Records: 0  Duplicates: 0  Warnings: 0

MariaDB [icingaweb]> ALTER TABLE `icingaweb_group_membership` CONVERT TO CHARACTER SET utf8mb4;
ERROR 1709 (HY000): Index column size too large. The maximum column size is 767 bytes.
MariaDB [icingaweb]>

**Comment by lippserd** [comment]
You have to add this to your MySQL config:

```
innodb_file_format=barracuda
innodb_file_per_table=1
innodb_large_prefix=1

**Comment by tomabg** [comment]
just added this settings..rebooted and still get the same issue

#  cat /etc/my.cnf |grep inno
innodb_file_format=barracuda
innodb_file_per_table=1
innodb_large_prefix=1
# mysql -u root -p icingaweb < /usr/share/doc/icingaweb2/schema/mysql-upgrades/2.11.0.sql
Enter password:
ERROR 1071 (42000) at line 2: Specified key was too long; max key length is 767 bytes
#

**Comment by lippserd** [comment]
Where exactly did you add this settings? It has to be beneath `[mysqld]`.

**Comment by tomabg** [comment]
#cat /etc/my.cnf
[mysqld]
innodb_file_format=barracuda
innodb_file_per_table=1
innodb_large_prefix=1
datadir=/var/lib/mysql
socket=/var/lib/mysql/mysql.sock
# Disabling symbolic-links is recommended to prevent assorted security risks
symbolic-links=0
# Settings user and group are ignored when systemd is used.
# If you need to run mysqld under a different user or group,
# customize your systemd unit file for mariadb according to the
# instructions in http://fedoraproject.org/wiki/Systemd

[mysqld_safe]
log-error=/var/log/mariadb/mariadb.log
pid-file=/var/run/mariadb/mariadb.pid


#
# include all files from the config directory
#
!includedir /etc/my.cnf.d

#mysql -u root -p icingaweb < /usr/share/doc/icingaweb2/schema/mysql-upgrades/2.11.0.sql
Enter password:
ERROR 1709 (HY000) at line 2: Index column size too large. The maximum column size is 767 bytes.
#

**Comment by mwaldmueller** [comment]
Had the same, this helped:
```
mysql> ALTER TABLE icingaweb_group_membership ROW_FORMAT=DYNAMIC;
mysql> ALTER TABLE icingaweb_user ROW_FORMAT=DYNAMIC;
mysql> ALTER TABLE icingaweb_user_preference ROW_FORMAT=DYNAMIC;
```

**Comment by lippserd** [comment]
Just also add `innodb_default_row_format=DYNAMIC` to the MySQL config.

**Comment by mwaldmueller** [comment]
> @mwaldmueller Did you test with `innodb_large_prefix=1`?

Yes, but that didn't change anything. I even re-created  database and tables during the process. Setting the "row_format" was mandatory.

**Comment by lippserd** [comment]
Ok `innodb_default_row_format` is only available in later versions 😆

**Comment by tomabg** [comment]
ok fixed for me...would you like to create a hotfix or close this issue with the Workarounds?

**Comment by lippserd** [comment]
> ok fixed for me...would you like to create a hotfix or close this issue with the Workarounds?

Let's leave the issue open.
